### PR TITLE
Implement binary and JSON encoders/decoders

### LINKED ISSUE #168: Simplify structure unmarshaling for SDK users
There is an straightforward way to serialize SDK structure by converting it into unified structure and then using stable marshaler on it.

```go
data, err := bearerToken.ToV2().StableMarshal(nil)
```

But there is not simple way to convert it back. To do that, user has to use transport structure even though it is not necessary, because unified structures are matched with protobuf definition. 

```go
grpcToken := new(grpcACL.BearerToken)
err = grpcToken.Unmarshal(data)  // we can omit that

v2Token := v2ACL.BearerTokenFromGRPCMessage(grpcToken)
bearerToken := token.NewBearerTokenFromV2(v2Token)
```

The issue is that unlike stable marshallers, unmarshallers are a little more complicated. And we don't have to make them stable (with strict field order check in binary message) at all. Can we just hide GRPC or any other transport layer behind `Unmarshal()` function in `v2/*` packages?

### PR DESCRIPTION
**Pull request by cthulhu-rider**
Closes #168.
### Netmap
- [x] Filter

- [x] Selector

- [x] Replica

- [x] PlacementPolicy

- [x] NodeInfo.Attribute

- [x] NodeInfo
### Accounting

- [x] Decimal
### ACL

- [x] EACLRecord.Filter

- [x] EACLRecord.Target

- [x] EACLRecord

- [x] EACLTable

- [x] BearerToken.Body.TokenLifetime

- [x] BearerToken.Body

- [x] BearerToken
### Container

- [x] Container.Attribute

- [x] Container
### Object

- [x] ShortHeader

- [x] Header.Attribute

- [x] Header.Split

- [x] Header

- [x] HeaderWithSignature

- [x] Object
### Refs

- [x] Address

- [x] ObjectID

- [x] ContainerID

- [x] OwnerID

- [x] Version

- [x] Signature

- [x] Checksum
### Session

- [x] ObjectSessionContext

- [x] SessionToken.Body.TokenLifetime

- [x] SessionToken.Body

- [x] SessionToken

- [x] XHeader

- [x] RequestMetaHeader

- [x] ResponseMetaHeader

- [x] RequestVerificationHeader

- [x] ResponseVerificationHeader
### Storagegroup

- [x] StorageGroup

### CODE REVIEW

File: v2/netmap/marshal.go [hunk 1]
```diff
@@ -118,35 +131,35 @@ func (s *Selector) StableMarshal(buf []byte) ([]byte, error) {
...
-	n, err = proto.StringMarshal(attributeSelectorField, buf[offset:], s.attribute)
+	n, err = protoutil.StringMarshal(attributeSelectorField, buf[offset:], s.attribute)
 	if err != nil {
 		return nil, err
 	}
 
 	offset += n
 
-	n, err = proto.StringMarshal(filterSelectorField, buf[offset:], s.filter)
+	n, err = protoutil.StringMarshal(filterSelectorField, buf[offset:], s.filter)
```
Comment (im-kulikov) [inline v2/netmap/marshal.go:162]: `n` can be omitted (`_, err = `)

File: v2/netmap/marshal.go [hunk 2]
```diff
@@ -178,14 +191,14 @@ func (r *Replica) StableMarshal(buf []byte) ([]byte, error) {
...
-	n, err = proto.UInt32Marshal(countReplicaField, buf[offset:], r.count)
+	n, err = protoutil.UInt32Marshal(countReplicaField, buf[offset:], r.count)
 	if err != nil {
 		return nil, err
 	}
 
 	offset += n
 
-	n, err = proto.StringMarshal(selectorReplicaField, buf[offset:], r.selector)
+	n, err = protoutil.StringMarshal(selectorReplicaField, buf[offset:], r.selector)
```
Comment (im-kulikov) [inline v2/netmap/marshal.go:201]: `n` can be omitted (`_, err = `)

File: v2/netmap/marshal.go [hunk 3]
```diff
@@ -338,30 +351,30 @@ func (ni *NodeInfo) StableMarshal(buf []byte) ([]byte, error) {
...
+		n, err = protoutil.NestedStructureMarshal(attributesNodeInfoField, buf[offset:], ni.attributes[i])
 		if err != nil {
 			return nil, err
 		}
 
 		offset += n
 	}
 
-	n, err = proto.EnumMarshal(stateNodeInfoField, buf[offset:], int32(ni.state))
+	n, err = protoutil.EnumMarshal(stateNodeInfoField, buf[offset:], int32(ni.state))
```
Comment (im-kulikov) [inline v2/netmap/marshal.go:377]: `n` can be omitted (`_, err = `)

**Review by im-kulikov — DISMISSED** [review]
`n` can be omitted (`_, err = `) if you don't need it

File: v2/netmap/marshal.go (hunk 1 above)
Comment (cthulhu-rider) [reply v2/netmap/marshal.go:162]: Please, don't repeat linter messages. `n` is outside this PR.

Comment (im-kulikov) [reply v2/netmap/marshal.go:162]: oh, not seen GA/lint notifycation

**Review by alexvanin — APPROVED** [review]

**Review by alexvanin — APPROVED** [review]

**Pull request merged**